pkg/adapter/pods.go

```diff
@@ -683,25 +683,27 @@ func kubeContainerToCreateConfig(ctx context.Context, containerYAML v1.Container
 		containerConfig.User = imageData.Config.User
 	}
 
-	if containerConfig.SecurityOpts != nil {
-		if containerYAML.SecurityContext.ReadOnlyRootFilesystem != nil {
-			containerConfig.ReadOnlyRootfs = *containerYAML.SecurityContext.ReadOnlyRootFilesystem
-		}
-		if containerYAML.SecurityContext.Privileged != nil {
-			containerConfig.Privileged = *containerYAML.SecurityContext.Privileged
-		}
+	if containerYAML.SecurityContext != nil {
```

Comment:
Could have made this a lot simpler by combining these two if commands on same line.

Reply:
for some reason, the AllowPRivilegeEscalation check was outside of the containerConfig.SecurityOpts check, so doing it this way maintained the branch structure. I didn't know enough about why that happened to choose to change it